recipes/dd-handler.rb

```diff
@@ -65,9 +65,14 @@ def extra_endpoints
   def handler_config
     extra_config = node['datadog']['handler_extra_config'].reject { |_, v| v.nil? }
 
-    # Since Agent 6 supports node['datadog']['url'] = nil, we need to fallback
-    # on a default value here.
+    # Since Agent 6 supports node['datadog']['url'] = nil and node['datadog']['site'] = nil,
+    # we need to fallback on a default value here.
+
+    # Passing the site option directly to the handler is only supported since v0.12.0 of chef-handler-datadog.
+    # To preserve compatibility with older versions, the site option is therefore
+    # resolved here, and only the URL is passed.
     dd_url = 'https://app.datadoghq.com'
+    dd_url = 'https://app.' + node['datadog']['site'] unless node['datadog']['site'].nil?
```

Comment:
If you set both, the Agent gives a Warning on startup, so users should know this is incorrect. Since this uses the settings that are set for the Agent, I think it's okay to not warn here (as the Agent will warn already).

Reply:
> Should we document it somewhere that site has precedence over the url ?
> Not likely to happen, but what if both site and url are set ?

If both are set, the url takes precedence (same as what we do for the config): https://github.com/DataDog/chef-datadog/blob/252589be4b372ab759cf847140f97457ccef2f97/recipes/_agent6_config.rb#L26-L30

This behavior is already documented in the attributes configuration: https://github.com/DataDog/chef-datadog/blob/252589be4b372ab759cf847140f97457ccef2f97/attributes/default.rb#L96-L97